sdk/core/src/index.ts

```diff
@@ -590,62 +461,27 @@ var ReactMoE = {
    * @param {JSONObject} pushPayload
    */
   passFcmPushPayload: function (pushPayload: object) {
-    commonValidationCheck();
     MoEngageLogger.verbose("Will process push payload.");
-    let payload = getMoEPushCampaignJson(pushPayload, PUSH_SERVICE_FCM, moeAppId);
-
     if (Platform.OS == PLATFORM_ANDROID) {
-      MoERNAndroid.passFcmPushPayload(payload);
-    } else {
-      MoEngageLogger.debug("This api is not supported on iOS platform.");
+      let payload = getMoEPushCampaignJson(pushPayload, PUSH_SERVICE_FCM, moeAppId);
+      MoEReactBridge.passFcmPushPayload(payload);
     }
   },
 
   /**
    * Call this method to register for push notification in iOS
    */
   registerForPush: function () {
-    commonValidationCheck();
-    MoEngageLogger.verbose("Will registerForPush");
-    if (Platform.OS == PLATFORM_ANDROID) {
-      MoEngageLogger.debug("This api is not supported on android platform.");
-    } else if (Platform.OS == PLATFORM_IOS) {
-      MoERNiOS.registerForPush();
-    }
-  },
-
-  /**
-   * Call this method to disable Inbox/ Notification Center feature.
-   */
-  disableInbox: function () {
-    commonValidationCheck();
-    MoEngageLogger.verbose("Will disableInbox");
-    if (Platform.OS == PLATFORM_ANDROID) {
-      MoEngageLogger.debug("This api is not supported on android platform.");
-    } else if (Platform.OS == PLATFORM_IOS) {
-      MoERNiOS.disableInbox(getAppIdJson(moeAppId));
-    }
-  },
-
-  /**
-   * Call this method to start Geofence tracking, this method also asks for location permission if not already done
-   */
-  startGeofenceMonitoring() {
-    if (Platform.OS == PLATFORM_ANDROID) {
-      //Android code
-    } else if (Platform.OS == PLATFORM_IOS) {
-      MoERNiOS.startGeofenceMonitoring(getAppIdJson(moeAppId));
+    if (Platform.OS == PLATFORM_IOS) {
+      MoEngageLogger.verbose("Will registerForPush");
+      MoEReactBridge.registerForPush();
     }
   },
 
   optOutDataTracking: function (shouldOptOutDataTracking: boolean) {
```

Comment:
can we remove this method and have enableDataTracking and disabledDataTracking?
It will be consistent with other platforms. Since it is a major release we can make this breaking change.

Reply:
@umangmoe done